packages/core/src/VDocument.ts

```diff
@@ -101,8 +97,7 @@ export class VDocument {
      * Truncate the tree by removing the selected nodes and merging their
      * orphaned children into the parent of the first removed node.
      */
-    deleteSelection(): void {
-        const range = this.selection.range;
+    deleteSelection(range: VRange): void {
```

Comment:
Why not use the document's selection's range by default if nothing is passed?

Reply:
It's an opinionated choice, actually. This one is not a command per se, you can't `execCommand` it. It is merely a "public helper" that is called by the other commands. If `range` is not required, you can write this:
```
function myCommand(params: CommandParams) {
    const range = params.range || selection.range;
    [do stuff]
    [do more stuff]
    [do yet more stuff, your brain hurts]
    this.deleteSelection();
    [so much stuff]
}
```
... and you just introduced a bug if a range is passed to your function, because you are not forwarding it to `deleteSelection`.

Reply:
Ok right that makes sense.